pkg/kubectl/cmd/expose.go

```diff
@@ -175,6 +176,9 @@ func RunExpose(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []str
 		}
 		params["labels"] = kubectl.MakeLabels(labels)
 	}
+	if cmdutil.GetFlagBool(cmd, "headless") {
```

Comment:
You don't actually need to add this. Just add "headless" as a parameter in  pkg/kubectl/service.go#paramNames and [MakeParams](https://github.com/kubernetes/kubernetes/blob/4566e039bf52a18c34b03c926541738be61957c7/pkg/kubectl/generate.go#L67) will do the rest for you.